meilisearch-release-changelog/v1.8.0.md

```diff
@@ -0,0 +1,219 @@
+
+### New embedders: generic REST embedder and Ollama model
+
+New embedder sources have been added
+- `ollama` source
+- `rest` source
+
+**REST embedder**
+
+Meilisearch now supports any REST embedder. You can set them up with the following configuration:
+
+```json5
+"default": {
+  "source": "rest", // 👈 Use the REST source
+  "url": "http://localhost:12345/api/v1/embed",
+  // ☝️ Mandatory, full URL to the embedding endpoint
+  "apiKey": "187HFLDH97CNHN",
+  // ☝️ optional, will be passed as Bearer in the Authorization header
+  "dimensions": 512,
+  // ☝️ optional, inferred with a dummy request if missing
+  "documentTemplate": "blabla",
+  "inputField": ["data", "text"],
+  // ☝️ inject texts in data.text in the query
+  // Optional, defaults to []
+  "inputType": "text", // text or textArray
+  // ☝️ inject a single text
+  // Optional, defaults to text
+  "query": {
+    // A JSON object describing other fields to send in a query
+    // for example
+    "model": "name-of-your-model",
+    "dimensions": 512
+  },
+  // ☝️ A JSON object describing other fields to send in a query
+  // Optional, defaults to {}
+  "pathToEmbeddings": ["data"],
+  // ☝️ look at embeddings in "data" in the response
+  // Optional, defaults to []
+  "embeddingObject": ["embedding"]
+  // ☝️ look at the embedding inside of "embedding"
+  // Optional, defaults to []
+}
+```
+
+Here is an example of setting OpenAI embedder with the `rest` source:
+
+```json5
+{
+  "source": "rest",
+  "apiKey": "<your-openai-api-key>",
+  "dimensions": 1536,
+  "url": "https://api.openai.com/v1/embeddings",
+  "query": {
+    "model": "text-embedding-ada-002"
+  },
+  "inputField": ["input"],
+  "inputType": "textArray",
+  "pathToEmbeddings": ["data"],
+  "embeddingObject": ["embedding"]
+}
+```
+
+**Ollama model**
+
+Here is how to set up the Ollama model:
+
+```json5
+"default": {
+  "source": "ollama", // 👈 Use the Ollama source
+  "url": "http://localhost:11434/api/embeddings",
+  // ☝️ optional, fetched from MEILI_OLLAMA_URL environment variable if missing
+  "apiKey": "<foobarbaz>",
+  // ☝️ optional
+  "model": "nomic-embed-text",
+  "documentTemplate": "blabla" // like for openAI and huggingFace sources
+}
+
+```
+
+### Expose the `distribution` shift setting
+
+When setting an embedder, you can now set the `distribution` shift.
+
+```json5
+"default": {
+  "source": "huggingFace", // supported for any source
+  "model": "some/model",
+  "distribution": {  // describes the natural distribution of results
+    "mean": 0.7, // mean value
+    "sigma": 0.3 // variance
+  }
+}
+```
+
+The “distribution shift” is an affine transformation applied to the `_rankingScore` of a **semantic** search result with the aim of making the comparison to the `_rankingScore` of a **keyword** search result more meaningful.
```

Comment:
````suggestion
### New embedder setting: `distribution`

Use `distribution` to apply an affine transformation to the `_rankingScore` of **semantic** search results. This can help to compare `_rankingScore`s of semantic and keyword search results and improve result ranking.

```json5
"default": {
  "source": "huggingFace",
  "model": "MODEL_NAME",
  "distribution": {  // describes the natural distribution of results
    "mean": 0.7, // mean value
    "sigma": 0.3 // variance
  }
}
```
````